deltacat/compute/compactor/steps/hash_bucket.py

```diff
@@ -203,7 +207,7 @@ def _timed_hash_bucket(
             num_buckets,
```

Comment:
can we also ensure the grouping has retained the same number of records? I think instead of the above assertion (where we have change return types from all the methods), this would be more useful. What do you think?

Reply:
The number of records aggregated up to the driver are still used to assert the expected number of materialized records in the parent workflow (after subtracting deduped records), so I think this would be sufficient for now.